### PR TITLE
Merge wxAdvanced into wxCore

### PR DESCRIPTION
**Pull request by vadz**
AFAICS this is the minimum required. With these changes, I can build wxGTK, wxX11 (i.e. wxUniv build) and wxMSW as DLLs.

Please let me know if you see anything I forgot or if you disagree with the `dummy.cpp` hack (I'm on a fence about this one, on one hand it's ugly, but OTOH it doesn't seem worth introducing another dummy source file just for this, how much dumminess do we want to have).

### CODE REVIEW

**Comment by MaartenBent** [comment]
There is some mentioning if `adv` in
`misc\scripts\ctags.ignore`
`docs\doxygen\mainpages\screenshots.h`
`docs\doxygen\mainpages\const_cpp.h`
And a lot of `@library{wxadv}` in the interface headers.

**Comment by vadz** [comment]
Good points, thanks @MaartenBent, will fix.

File: build/cmake/samples/CMakeLists.txt [hunk 1]
```diff
@@ -8,22 +8,22 @@
 #############################################################################
 
 wx_add_sample(access accesstest.cpp DEPENDS wxUSE_ACCESSIBILITY)
-wx_add_sample(animate anitest.cpp anitest.h LIBRARIES adv DATA throbber.gif hourglass.ani DEPENDS wxUSE_ANIMATIONCTRL)
+wx_add_sample(animate anitest.cpp anitest.h DATA throbber.gif hourglass.ani DEPENDS wxUSE_ANIMATIONCTRL)
```
[the remark this answers is not in the document]
Comment (vadz) [reply build/cmake/samples/CMakeLists.txt]: If I understand CMake syntax correctly (a big if, admittedly), `LIBRARIES` is a keyword and so is not needed (or maybe even harmful?) if there is nothing following it.

[the remark this answers is not in the document]
Comment (MaartenBent) [reply build/cmake/samples/CMakeLists.txt]: Not harmful, but other optional keywords are omitted as well when they have no values.

**Pull request merged**